src/constants.ts

```diff
@@ -4,7 +4,17 @@ export const DISPLAY_INTERVAL_DEFAULT = 30000;
 /* how many times we try to create a new user when _setUserID_ is invoked */
 export const RETRY_USER_ATTEMPTS = 0;
 
-export const BASE_URL = process.env.BASE_URL || 'https://api.iterable.com/api';
+const IS_EU_SERVICE = process.env.IS_EU_SERVICE === 'true' ? true : false;
+
+const US_ITERABLE_DOMAIN = 'api.iterable.com';
+
+const EU_ITERABLE_DOMAIN = 'eu.iterable.com';
+
+const ITERABLE_API_URL = `https://${
+  IS_EU_SERVICE ? EU_ITERABLE_DOMAIN : US_ITERABLE_DOMAIN
+}/api`;
+
+export const BASE_URL = process.env.BASE_URL || ITERABLE_API_URL;
```

Comment:
@mprew97 Even if `IS_EU_SERVICE` is true, the `BASE_URL` env variable here will be used and the EU domain will just be the fallback.

What is the upside of allowing users to set an `IS_EU_SERVICE` variable as well as a `BASE_URL` variable? Would it suffice to just allow users to set `BASE_URL` in the env file to "https://eu.iterable.com/api" (or the US domain depending on their use)?

Reply:
To your first point, I would say if they want a custom URL, whether for development or whatever, then that custom URL should take priority. Maybe that indicates `BASE_URL` should be `CUSTOM_URL`.

To the second, `BASE_URL` could be a fallback instead if we don't anticipate users ever having custom URLs. It's really based on whatever is a better developer experience. I feel that at least just allowing them to set flags for their preferred env is a better experience since devs don't have to worry about typos/wrong urls, they can just toggle a flag for their preferred region. I'm also not sure if there's a need to specify a subdomain since all of the endpoints hit the same subdomain `api.iterable.com` for the time being. If that ever changes, we could extend it. 

Thoughts? @pauljung14

Reply:
@pauljung14 and I had a conversation around this behavior. Since `BASE_URL` still currently takes precedence, in order to make sure there are no breaking changes, we are going to leave the env var in, change the flag name to `IS_ITERABLE_EU_SERVICE`, and leave some comments about the priority of the environment variables.